roadmap.md

```diff
@@ -0,0 +1,44 @@
+# WebDriver BiDi roadmap
+
+## Real-world end-to-end user scenarios
+
+This document presents an overview of real-world end-to-end user scenarios we aim to enable via the WebDriver BiDi protocol. Each scenario requires one or more WebDriver BiDi commands and events to be specified, tested, and implemented across browser engines.
+
+### A. Extracting headlines
+
+This scenario loads a web page and uses script evaluation to extract the headlines on the page. In spec terms, this involves:
+
+- [x] [handling sessions](https://w3c.github.io/webdriver-bidi/#module-session)
+- [x] [creating a new `browsingContext`](https://w3c.github.io/webdriver-bidi/#command-browsingContext-create)
+- [x] [navigating to a URL](https://w3c.github.io/webdriver-bidi/#command-browsingContext-navigate)
+- [x] [evaluating JavaScript in the page context](https://w3c.github.io/webdriver-bidi/#command-script-evaluate)
+- [x] [serialization/deserialization of JavaScript values](https://w3c.github.io/webdriver-bidi/#data-types-protocolValue)
+
+### B. Submitting forms
+
+This scenario loads a web page, enter text into a form field via the keyboard, and submits the form via a mouse click before extracting the results from the page. In spec terms, this involves:
+
+- [x] (everything from the previous milestone)
+- [ ] [emulating keyboard input](https://github.com/w3c/webdriver-bidi/pull/175)
+- [ ] [emulating mouse input](https://github.com/w3c/webdriver-bidi/pull/175)
+
+### C. Capturing screenshots
+
+This scenario loads a web page and captures a screenshot. In spec terms, this involves:
+
+- [x] (some items from the previous milestones)
+- [ ] [capturing a screenshot via a command](https://w3c.github.io/webdriver-bidi/#command-browsingContext-captureScreenshot)
+
+### D. Printing to PDF
+
+This scenario loads a web page and prints it as a PDF. In spec terms, this involves:
+
+- [x] (some items from the previous milestones)
+- [ ] [printing to PDF via a command](https://github.com/w3c/webdriver-bidi/issues/210)
+
+### E. Blocking images
```

Comment:
HAR file generation => network logging?

Reply:
Yes, agreed. For the spec, and implementations not based on CDP, network request logging is a necessary precursor to network request interception, and opens up clear use cases which don't require interception; notably simple correctness checks around ensuring that a request completes as expected, and request timing / HAR generation as part of performance logging. We've had specific requests for the latter of these from (iirc) BrowserTime.

Reply:
@OrKoN does that answer your question? Can this issue be closed?

Reply:
My suggestion was to mention the HAR generation use case and link to the network logging feature. It can be done in a follow-up though.

Reply:
Thanks. Let’s pursue this a follow-up then.

Reply:
Note that I'll have a network logging example in my upcoming proposal. But it will exclude the HAR file generation given that this is not part of the BiDi spec.

Reply:
I don't think to be able to mark this as resolved: please consider this thread resolved.